lib/ansible/modules/packaging/os/yum.py

```diff
@@ -390,6 +391,44 @@ def __init__(self, module):
         self.pkg_mgr_name = "yum"
         self.lockfile = '/var/run/yum.pid'
 
+    def is_lockfile_pid_valid(self):
+        try:
+            with open(self.lockfile, 'r') as f:
+                oldpid = int(f.readline())
+        except ValueError:
+            # invalid data
+            os.unlink(self.lockfile)
+            return False
+        except (IOError, OSError) as e:
+            self.module.fail_json(msg="Failure opening %s: %s" % (self.lockfile, to_native(e)))
+
+        if oldpid == os.getpid():
+            # that's us?
+            os.unlink(self.lockfile)
+            return False
+
+        try:
+            with open("/proc/%d/stat" % oldpid, 'r') as f:
+                stat = f.readline()
+
+            if stat.split()[2] == 'Z':
+                # Zombie
+                os.unlink(self.lockfile)
+                return False
+        except IOError:
+            try:
+                os.kill(oldpid, 0)
```

Comment:
Is it safe to kill the oldpid on IOError?  Under what circumstances is IOError raised?

Reply:
> Is it safe to kill the oldpid on IOError?

Not 'kill' but send `0` signal to just perform the error checking.

> Under what circumstances is IOError raised?

IIRC either `proc` is not mounted or pid is dead.